sortOddNumber.js

```diff
@@ -7,29 +7,21 @@ Examples
 [9, 8, 7, 6, 5, 4, 3, 2, 1, 0]  =>  [1, 8, 3, 6, 5, 4, 7, 2, 9, 0]
 */
 
-function sortArray(array) {
-    // Return a sorted array.
-      let stack = []
-      for(let i=0; i< array.length;i++){
-        let currentPosition = i
-        if(array[i] %2 !== 0){
-          console.log(i)
-          for(let j = stack.length-1;j>=0;j--){
-                let prePosition = stack[j]
-                if(array[currentPosition] < array[prePosition]){
-                   let temp = array[currentPosition]
-                   array[currentPosition] = array[prePosition]
-                   array[prePosition] = temp
-                   currentPosition = prePosition
-                }
-                else{
-                   break
-                }
-          }
-          stack.push(i) // add the postion of odd number to the stack  
+var sortArray = (array) => {
+  // Return a sorted array.
+  let sortedArray = array.reduce((previousValue, currentValue) => {
+    previousValue.push(currentValue)
+    if (currentValue % 2 !== 0) {
+      let currentIndex = previousValue.length - 1
+      for (let i = previousValue.length - 2; i >= 0; i--) {
```

Comment:
there is an easier way to solve this using just a regular for loop and having to go through the array once making it O(n). this would be O(n^2) since worse case you are going through the whole array on every iteration

Reply:
Hi Don, I'm not sure how we can use 1 iteration to make it O(n) for the sorting.
There is another way that get all the odd numbers and sort into 1 array and put them back to the Array. and this way I think we process the array twice => 2n which better than n^2.
What do you think about this?

Reply:
use the best algorithm you can think of to resolve this